main/test/flix/Test.Exp.Match.Record.flix

```diff
@@ -0,0 +1,34 @@
+namespace Test/Exp/Match/Record {
+
+    @test
+    def testMatchRecord01(): Bool = match {x = 12, y = 10} {
+        case { x, y } => x == 12 && y == 10
+        case _ => false
+    }
+
+    @test
+    def testMatchRecord01(): Bool = match {x = "a", y = "b"} {
+        case { x, y } => x == "a" && y == "b"
+        case _ => false
+    }
+
+    @test
+    def testMatchRecord01(): Bool = match {x = 12, y = 10} {
+        case { x } => x == 12
+        case _ => false
+    }
+
+    @test
+    def testMatchRecord01(): Bool = match {x = 12, y = 10} {
+        case { y } => y == 10
+        case _ => false
+    }
+
+    @test
+    def testMatchRecord01(): Bool = match {x = 12, y = 10} {
+        case { y } => y == 10
+        case { x } => false
+        case _ => false
+    }
+
```

Comment:
Looks like a reasonable start. You also want some test cases where the order is swapped, eg:

```
    def testMatchRecord01(): Bool = match {y = 10, x = 12} {
        case { x, y } => x == 12 && y == 10
        case _ => false
    }
```

And I think you can even have test cases without a default case:

```
    def testMatchRecord01(): Bool = match {y = 10, x = 12} {
        case { x, y } => x == 12 && y == 10
        case _ => false
    }
```

Reply:
Oh, and you also want some test cases with nested patterns and nested records! :)

Reply:
Cool, thanks!

Reply:
The second test case should use y right? Also I think that we will need to check for the exhaustiveness then :)

Reply:
Can we have something like this as nested Record, right?

```
{ x = 10, y = { x = 11, z = 12 } }
```

In this case I think we would also like to extend the syntax to allow renaming, something like:

```
case { x, y = { x as x1, z } } =>
```

Reply:
> The second test case should use y right? Also I think that we will need to check for the exhaustiveness then :)

No, I don't have to. We have extensible records, so I should not be required to use a field just because it is there.

Reply:
For renaming, I would leave that for a discussion later.

Reply:
Maybe we should even start without nested records. (for simplicity)

Reply:
> > The second test case should use y right? Also I think that we will need to check for the exhaustiveness then :)
> 
> No, I don't have to. We have extensible records, so I should not be required to use a field just because it is there.

Right :)